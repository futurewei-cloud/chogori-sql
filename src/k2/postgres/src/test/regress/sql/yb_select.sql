--
-- SELECT
--

CREATE TABLE nr (i int, j int, PRIMARY KEY (j ASC));
CREATE INDEX ON nr (i ASC);
INSERT INTO nr VALUES (1, 2), (3, 4);
BEGIN;
INSERT INTO nr VALUES (null, 6);
SELECT i FROM nr ORDER BY i;
SELECT i FROM nr ORDER BY i NULLS FIRST;
SELECT i FROM nr ORDER BY i DESC;
SELECT i FROM nr ORDER BY i DESC NULLS LAST;
ROLLBACK;

CREATE TABLE nr2 (i int, j int, PRIMARY KEY (j ASC));
CREATE INDEX ON nr2 (i DESC);
INSERT INTO nr2 VALUES (1, 2), (3, 4);
BEGIN;
INSERT INTO nr2 VALUES (null, 6);
SELECT i FROM nr2 ORDER BY i;
SELECT i FROM nr2 ORDER BY i NULLS FIRST;
SELECT i FROM nr2 ORDER BY i DESC;
SELECT i FROM nr2 ORDER BY i DESC NULLS LAST;
ROLLBACK;

--
-- Test reverse scans with limit.
--
CREATE TABLE reverse_scan_test (
  h BIGINT,
  r INT,
  PRIMARY KEY(h, r ASC)
);

INSERT INTO reverse_scan_test
VALUES (1, 1),
       (1, 2),
       (1, 3),
       (1, 4),
       (1, 5),
       (1, 6),
       (1, 7),
       (1, 8),
       (1, 9),
       (1, 10),
       (1, 11),
       (1, 12),
       (1, 13),
       (1, 14);

-- Check ascending order.
SELECT r FROM reverse_scan_test WHERE h = 1;

SELECT r FROM reverse_scan_test WHERE h = 1 AND r > 2 and r < 9;

SELECT r FROM reverse_scan_test WHERE h = 1 AND r > 2 and r < 9 LIMIT 5;

SELECT r FROM reverse_scan_test WHERE h = 1 AND r >= 2 and r <= 13;

SELECT r FROM reverse_scan_test WHERE h = 1 AND r >= 2 and r <= 13 LIMIT 10;

-- Test desc order (reverse scan).
SELECT r FROM reverse_scan_test WHERE h = 1 ORDER BY r DESC;

SELECT r FROM reverse_scan_test WHERE h = 1 ORDER BY r DESC LIMIT 5;

SELECT r FROM reverse_scan_test WHERE h = 1 AND r < 9 ORDER BY r DESC LIMIT 5;

SELECT r FROM reverse_scan_test WHERE h = 1 AND r > 1 AND r < 14 ORDER BY r DESC;

SELECT r FROM reverse_scan_test WHERE h = 1 AND r > 1 AND r < 14 ORDER BY r DESC LIMIT 9;

SELECT r FROM reverse_scan_test WHERE h = 1 AND r >= 3 and r <= 13 ORDER BY r DESC;

SELECT r FROM reverse_scan_test WHERE h = 1 AND r >= 3 and r <= 13 ORDER BY r DESC LIMIT 8;
