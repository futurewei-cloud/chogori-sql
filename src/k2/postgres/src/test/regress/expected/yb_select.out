--
-- SELECT
--
CREATE TABLE nr (i int, j int, PRIMARY KEY (j ASC));
CREATE INDEX ON nr (i ASC);
INSERT INTO nr VALUES (1, 2), (3, 4);
BEGIN;
INSERT INTO nr VALUES (null, 6);
SELECT i FROM nr ORDER BY i;
 i
---
 1
 3

(3 rows)

SELECT i FROM nr ORDER BY i NULLS FIRST;
 i
---

 1
 3
(3 rows)

SELECT i FROM nr ORDER BY i DESC;
 i
---

 3
 1
(3 rows)

SELECT i FROM nr ORDER BY i DESC NULLS LAST;
 i
---
 3
 1

(3 rows)

ROLLBACK;
CREATE TABLE nr2 (i int, j int, PRIMARY KEY (j ASC));
CREATE INDEX ON nr2 (i DESC);
INSERT INTO nr2 VALUES (1, 2), (3, 4);
BEGIN;
INSERT INTO nr2 VALUES (null, 6);
SELECT i FROM nr2 ORDER BY i;
 i
---
 1
 3

(3 rows)

SELECT i FROM nr2 ORDER BY i NULLS FIRST;
 i
---

 1
 3
(3 rows)

SELECT i FROM nr2 ORDER BY i DESC;
 i
---

 3
 1
(3 rows)

SELECT i FROM nr2 ORDER BY i DESC NULLS LAST;
 i
---
 3
 1

(3 rows)

ROLLBACK;
--
-- Test reverse scans with limit.
--
CREATE TABLE reverse_scan_test (
  h BIGINT,
  r INT,
  PRIMARY KEY(h, r ASC)
);
INSERT INTO reverse_scan_test
VALUES (1, 1),
       (1, 2),
       (1, 3),
       (1, 4),
       (1, 5),
       (1, 6),
       (1, 7),
       (1, 8),
       (1, 9),
       (1, 10),
       (1, 11),
       (1, 12),
       (1, 13),
       (1, 14);
-- Check ascending order.
SELECT r FROM reverse_scan_test WHERE h = 1;
 r
----
  1
  2
  3
  4
  5
  6
  7
  8
  9
 10
 11
 12
 13
 14
(14 rows)

SELECT r FROM reverse_scan_test WHERE h = 1 AND r > 2 and r < 9;
 r
---
 3
 4
 5
 6
 7
 8
(6 rows)

SELECT r FROM reverse_scan_test WHERE h = 1 AND r > 2 and r < 9 LIMIT 5;
 r
---
 3
 4
 5
 6
 7
(5 rows)

SELECT r FROM reverse_scan_test WHERE h = 1 AND r >= 2 and r <= 13;
 r
----
  2
  3
  4
  5
  6
  7
  8
  9
 10
 11
 12
 13
(12 rows)

SELECT r FROM reverse_scan_test WHERE h = 1 AND r >= 2 and r <= 13 LIMIT 10;
 r
----
  2
  3
  4
  5
  6
  7
  8
  9
 10
 11
(10 rows)

-- Test desc order (reverse scan).
SELECT r FROM reverse_scan_test WHERE h = 1 ORDER BY r DESC;
 r
----
 14
 13
 12
 11
 10
  9
  8
  7
  6
  5
  4
  3
  2
  1
(14 rows)

SELECT r FROM reverse_scan_test WHERE h = 1 ORDER BY r DESC LIMIT 5;
 r
----
 14
 13
 12
 11
 10
(5 rows)

SELECT r FROM reverse_scan_test WHERE h = 1 AND r < 9 ORDER BY r DESC LIMIT 5;
 r
---
 8
 7
 6
 5
 4
(5 rows)

SELECT r FROM reverse_scan_test WHERE h = 1 AND r > 1 AND r < 14 ORDER BY r DESC;
 r
----
 13
 12
 11
 10
  9
  8
  7
  6
  5
  4
  3
  2
(12 rows)

SELECT r FROM reverse_scan_test WHERE h = 1 AND r > 1 AND r < 14 ORDER BY r DESC LIMIT 9;
 r
----
 13
 12
 11
 10
  9
  8
  7
  6
  5
(9 rows)

SELECT r FROM reverse_scan_test WHERE h = 1 AND r >= 3 and r <= 13 ORDER BY r DESC;
 r
----
 13
 12
 11
 10
  9
  8
  7
  6
  5
  4
  3
(11 rows)

SELECT r FROM reverse_scan_test WHERE h = 1 AND r >= 3 and r <= 13 ORDER BY r DESC LIMIT 8;
 r
----
 13
 12
 11
 10
  9
  8
  7
  6
(8 rows)
